core/src/main/java/io/fabric8/maven/core/util/IoUtil.java

```diff
@@ -79,6 +88,33 @@ public static void download(Logger log, URL downloadUrl, File target) throws Moj
 
     }
 
+    /**
+     * Find a free (on localhost) random port in the range [49152, 65535] after 100 attempts.
+     *
+     * @return a random port where a server socket can be bound to
+     */
+    public static int getFreeRandomPort() {
+        // 100 attempts should be enough
+        return getFreeRandomPort(49152, 65535, 100);
+    }
+
+    /**
+     * Find a free (on localhost) random port in the specified range after the given number of attempts.
+     */
+    public static int getFreeRandomPort(int min, int max, int attempts) {
```

Comment:
Yes agree, might be even problematic since I'm not sure whether the socket is freed immediately after being closed. Maybe better to stick to your approach.